packages/grafana-data/src/utils/valueMappings.ts

```diff
@@ -58,9 +58,13 @@ export function getValueMappingResult(valueMappings: ValueMapping[], value: any)
 
         const regex = stringToJsRegex(vm.options.pattern);
         if (value.match(regex)) {
-          const thisResult = Object.create(vm.options.result);
-          thisResult.text = value.replace(regex, vm.options.result.text || '');
-          return thisResult;
+          const res = { ...vm.options.result };
+
+          if (res.text != null) {
```

Comment:
i think `!=` will also mean we don't go into this branch if `res.text === undefined`. is that what we want, or should this be `!==`? 🤔

Reply:
yes, the != is deliberate to also exclude undefined